### PR TITLE
Handling LoadBalancerIPStrings as a derived service

### PR DESCRIPTION
**Pull request by sridhartigera**
In case of LoadBalancers, the public IP is present in the LoadBalancerIPStrings in the service info. This is handled as a derived service and proper NAT entries are programmed for the LB public IP
1) This fixes the GCP loadbalancer issue
2) In case of aws, we get Loadbalancer domain names and not public IP. Hence we dont see a NAT entry for the resolved domain name in FE BPF map. The loadbalancer in turn does a DNAT of the <public ip:80> to <node_ip:nodeport>. Hence we will only nodeport NAT entries for aws. 

## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: bpf/proxy/lb_src_range_test.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ var _ = Describe("BPF Load Balancer source range", func() {
 				net.IPv4(10, 0, 0, 2),
 				2222,
 				v1.ProtocolTCP,
-				proxy.K8sSvcWithExternalIPs([]string{"35.0.0.2"}),
+				proxy.K8sSvcWithLoadBalancerIPs([]string{"35.0.0.2"}),
```
Comment (tomastigera) [inline bpf/proxy/lb_src_range_test.go:60]: I wonder if we should execute this test 2x, once for ExternalIPs and once for LBIPs

Comment (sridhartigera) [reply bpf/proxy/lb_src_range_test.go:60]: yes. Will add one more test for external IP

Comment (tomastigera) [reply bpf/proxy/lb_src_range_test.go:60]: We may just turn this set of tests into a function and run it 2x with 2 differnt options :thinking:

File: bpf/proxy/lb_src_range_test.go [hunk 2]
```diff
@@ -36,193 +36,212 @@ func init() {
...
+				err := s.Apply(state)
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(svcs.m).To(HaveLen(0))
+				Expect(eps.m).To(HaveLen(0))
+			}))
+
+		})
+	}
+	testfn([]string{}, []string{"35.0.0.2"})
```
Comment (tomastigera) [inline bpf/proxy/lb_src_range_test.go:245]: Since the two test cases are pretty much mutually exclusive, you could just pass the one option.

**Pull request merged**